src/emulator/storage/apis/gcloud.ts

```diff
@@ -35,6 +36,40 @@ export function createCloudEndpoints(emulator: StorageEmulator): Router {
       return;
     }
 
+    if (req.query.alt == "media") {
+      let data = storageLayer.getBytes(req.params.bucketId, req.params.objectId);
+      if (!data) {
+        res.sendStatus(404);
+        return;
+      }
+
+      const isGZipped = md.contentEncoding == "gzip";
+      if (isGZipped) {
+        data = gunzipSync(data);
+      }
+
+      res.setHeader("Accept-Ranges", "bytes");
+      res.setHeader("Content-Type", md.contentType);
+      res.setHeader("Content-Disposition", md.contentDisposition);
+      res.setHeader("Content-Encoding", "identity");
+
+      const byteRange = [...(req.header("range") || "").split("bytes="), "", ""];
+
+      const [rangeStart, rangeEnd] = byteRange[1].split("-");
+
+      if (rangeStart) {
+        const range = {
+          start: parseInt(rangeStart),
```

Comment:
nit: we may want to handle errors like NaN

Reply:
This is not handled on the [firebase API either](https://github.com/firebase/firebase-tools/blob/master/src/emulator/storage/apis/firebase.ts#L189-L195).

I'm not sure how you'd like to handle this. Would you mind opening a specific issue for this so it's treated separately that in this PR?
It feels like for the emulator these values should always be correct anyway, since the API calls are supposed to be made by the  `@google-cloud/storage` and `firebase` modules.